pkg/util/goroutinemap/goroutinemap.go

```diff
@@ -74,3 +74,28 @@ func (grm *goRoutineMap) operationComplete(operationName string) {
 	defer grm.Unlock()
 	delete(grm.operations, operationName)
 }
+
+// alreadyExistsError is specific error returned when NewGoRoutine()
+// detects that operation with given name is already running.
+type alreadyExistsError struct {
+	operationName string
+}
+
+var _ error = alreadyExistsError{}
+
+func (err alreadyExistsError) Error() string {
```

Comment:
nit: This method does not need to be public--it should only be called internally.